src/KubernetesClient/WatcherDelegatingHandler.cs

```diff
@@ -1,15 +1,23 @@
-using System;
-using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using System.Net;
 using System.Net.Http;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.WebUtilities;
 
 namespace k8s
 {
+    /// <summary>
+    /// Represents line separated http content returned from ?watch=true calls
+    /// </summary>
+    public interface ILineSeparatedHttpContent
```

Comment:
This should probably be marked internal because it is an internal implementation detail that most people shouldn't have to worry about.

Reply:
That makes it hard to unit test code that uses the Watch API.  With it exposed it's sufficient to do something like:

```csharp
mocks.k8s.ListNamespacedPodWithHttpMessagesAsync("test")
.Returns(Task.FromResult(new HttpOperationResponse<V1PodList>
{
    Body = new V1PodList(),
    Response = new HttpResponseMessage { Content = mockLineSeparatedHttpContent }
}));
```

Would putting it in an internal namespace (k8s.internal) suffice?  (asp.net core seems to use this pattern)

Reply:
You can use InternalsVisibleTo attribute to remove the issue entirely. I would check with the main code owners to see what their preference is.

Reply:
Sorry, I didn't mean for unit testing the KubernetesClient library itself, I meant for unit testing code that uses the KubernetesClient library.

Reply:
Ah, still, is this something people should be testing? It looks like it is deep enough in the code that people might be better off mocking a higher level abstraction.

Reply:
If you call watch, there's nothing you can mock -- you could e.g. write a thin wrapper:

```csharp
interface IWatcher 
{
  Task Watch<T>(HttpOperationResponse response, Action<> onEvent, ...)
}

class Watcher : IWatcher
{
  Task Watch<T>(HttpOperationResponse response, Action<> onEvent, ...) 
  {
    return response.Watch(onEvent, ...)
  }
}
```

would it be worth including something like that as an alternative to calling the extension function?